### ISSUE TITLE
Dot not escaped in routing

### ISSUE THREAD

**Issue by dugwood** [phalcon/cphalcon#2821]
As the manual states, we can use regular expressions for routing, such as:
http://docs.phalconphp.com/fr/latest/reference/routing.html#parameters-with-names

```php
    $router->add(
        "/documentation/{chapter}/{name}.{type:[a-z]+}",
        array(
            "controller" => "documentation",
            "action"     => "show"
        )
    );
```
But if you dump the route, you'll get:
```php
protected '_compiledPattern' => string '#^/documentation/([^/]*)/([^/]*).([a-z]+)$#' (length=43)
```
So the dot is not escaped, which can lead to valid urls such as:
/documentation/test/wordWdoc
as the dot means «any valid character».

Can you add escaping of dots? For now, the only solution I had was to add `{dot:\.}` which resolves as `(\.)`, and needs to be sent in the url dependency.

Thanks for this amazing tool :-)

**Comment by web777** [comment]
Use UTF8 codes. Dot is \x2E

**Comment by dugwood** [comment]
Sure, but to me that's still an issue, as I don't expect chars not in brackets to be interpreted.

First:
- /test.dotted => compile pattern is «/test.dotted», which is NOT a regexp
- /test.{dotted} => compile pattern is «#^/test.([^/]+)$#», which is a regexp

So, if there's no match in the rule, I can't use \x2E, because it would not be interpreted as it's not a regexp, just a string test (that's a good thing, as it's faster to parse the rule).

The main syntax of Phalcon router helps matching dedicated strings, hence the dot should be escaped as a default. Most people don't play nice with regexps, and the dot is one of the most common error.
I think it should be the opposite: «dot» should match a dot, and if you need the dot effect, use a placeholder, such as {name:[a-z]+}, which reduces the risk of a bad rule.

So your solution works, but this is a big tool used by a lot of people who don't go into the code as I do. We should protect their actions!

**Comment by andresgutierrez** [comment]
Fixed in 2.1.x

**Comment by dugwood** [comment]
Thanks @andresgutierrez .
